### PR TITLE
gmail validation

### PR DESCRIPTION
**Pull request by prachetos**

### CODE REVIEW

**Comment by JoshData** [comment]
Hi,

I'm open to adding gmail-specific checks, but we've got to get this into a much cleaner form first:

* No whitespace changes.
* No refactoring of the core code.
* No new module-level function. The gmail checks should be a part of `validate_email`, as a final step, if the domain is one that the gmail validator class is supposed to be used for.
* The new module should raise `EmailSyntaxError`, rather than defining a new exception class.
* The gmail-specific normalizations could be returned in a new key in the output dict.

We'll also need a way to test this so that we can be sure the rules don't incorrectly judge good addresses to be bad.

Thanks.

**Comment by prachetos** [comment]
Cool.. will get back to you. Your points are very valid.

File: README.rst [hunk 1]
```diff
@@ -383,3 +384,45 @@ them through the validator (without deliverability checks) like so:
...
+        }
+
+Key names:
+-----------
+::
+
+        email - original email
+        email_ascii - normalised email
+        local - normalised local part
+        domain - normalised domain which is always gmail.com
```
Comment (jab) [inline README.rst]: What about Google Mail for custom domains? I believe the same normalization rules apply there. It should be possible to tell when Google Mail is in use on a given domain based on the MX record (as mentioned in #3), in which case this feature could be applied there as well.

File: email_validator/__init__.py [hunk 2]
```diff
@@ -28,324 +29,360 @@
 
 # ease compatibility in type checking
 if sys.version_info >= (3,):
-	unicode_class = str
```
Comment (jab) [inline email_validator/__init__.py]: Looks like your editor changed the indentation of every .py file you included in this patch (granted to be PEP8 compliant, but still), it should probably be reworked to match the indentation of the existing code, so reviewers can focus on the functional changes.

@JoshData, you can use http://editorconfig.org, and tell contributors to use it as well, to catch this earlier.

**Pull request closed without merging**